### PR TITLE
Phone Number Authentication using Firebase in React Native

### LINKED ISSUE #1343: Phone Number Authentication using Firebase in a Non-Expo React Native Application
## Brief Summary
In this tutorial, we will develop a React Native application (Non-Expo) and add phone number authentication using Firebase.

## Key Takeaways

- Set up Firebase Project for Phone number authentication.
- Add 2 Screens: Log in and Authenticated screen.
- Use the Firebase Auth module and authenticate users using Phone numbers.
- Logout from the authenticated screen.

### PR DESCRIPTION
**Pull request by zolomohan**
### Article Checklist
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1343.

### CODE REVIEW

**Comment by katungi** [comment]
@hectorkambow, I would like to review this article

**Comment by hectorkambow** [comment]
@katungi  will be reviewing this article 🚀

**Comment by hectorkambow** [comment]
plagiarism check done 👀

**Comment by katungi** [comment]
@zolomohan this was an amazing tutorial. Very easy to follow through and the explanations and grammar were spot on.
@hectorkambow , I am done with both the grammar and tech review.

File: articles/react-native-firebase-phone-authentication/index.md [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+
+```JSX
+import React, { useState } from 'react';
+```
+
+```JSX
+const [confirm, setConfirm] = useState(null);
+```
+
+Now, let's write a function that accepts a phone number as an argument and calls the` signInWithPhoneNumber` function with the given phone number.
```
Comment (katungi) [inline articles/react-native-firebase-phone-authentication/index.md:245]: remove the tiny space before `signInWithPhoneNumber`

**Comment by zolomohan** [comment]
@katungi Thank you for reviewing this article.

**Comment by hectorkambow** [comment]
@katungi Please be sure to always add the PRL (peer reviewer line) at the bottom of articles to review - this it to ensure payout. 
Please be sure to also always indicate whether an article has been through a plagiarism check or not. (this is key - there is a label to use as well. )
👍  

Also please go over any small edits I made to implement into next reviews.

cc @zolomohan  for example using _ instead of * for italics. Its best to use* 

Thank you

**Pull request merged**